### PR TITLE
docs(glossary): Add more info about lossy/lossless compression

### PR DESCRIPTION
**Pull request by bsmth**
### Description

This PR fixes the Glossary entry for Lossless compression which has some see also links that are broken.

* linking back and forth between lossy and lossless pages
* mention that some compression methods have both lossy and lossless modes

### Motivation

Fix broken links, better xrefs

### Related issues and pull requests

- [x] https://github.com/mdn/content/pull/32465

### CODE REVIEW

File: files/en-us/glossary/gzip_compression/index.md [hunk 1]
```diff
@@ -6,9 +6,11 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-gzip is a file format used for file compression and decompression. It is based on the Deflate algorithm that allows files to be made smaller in size which allows for faster network transfers. gzip is commonly supported by web servers and modern browsers, meaning that servers can automatically compress files with gzip before sending them, and browsers can uncompress files upon receiving them.
+gzip is a file format used for file compression and decompression. It is based on [the Deflate algorithm](https://www.zlib.net/feldspar.html) that allows files to be made smaller in size which allows for faster network transfers. gzip is commonly supported by web servers and modern browsers, meaning that servers can automatically compress files with gzip before sending them, and browsers can uncompress files upon receiving them.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/gzip_compression/index.md:9]:
```suggestion
gzip is a file format used for file compression and decompression. It is based on [the Deflate algorithm](https://www.zlib.net/feldspar.html) which allows files to be made smaller in size, allowing for faster network transfers. gzip is commonly supported by web servers and modern browsers, meaning that servers can automatically compress files with gzip before sending them, and browsers can uncompress files upon receiving them.
```

File: files/en-us/glossary/lossless_compression/index.md [hunk 2]
```diff
@@ -6,15 +6,11 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-**Lossless compression** is a class of data compression algorithms that allows the original data to be perfectly reconstructed from the compressed data. Lossless compression methods are reversible. Examples of lossless compression include {{glossary("GZip_compression", "gzip")}}, {{glossary("Brotli_compression", "brotli")}}, {{glossary("WebP")}}, and {{glossary("PNG")}},
+**Lossless compression** is a class of data compression algorithms that allows the original data to be perfectly reconstructed from the compressed data. Lossless compression methods are reversible. Examples of lossless compression include {{glossary("GZip_compression", "gzip")}}, {{glossary("Brotli_compression", "brotli")}}, {{glossary("Zstandard compression", "Zstandard")}}, {{glossary("WebP")}}, and {{glossary("PNG")}}.
 
 {{glossary("Lossy compression")}}, on the other hand, uses inexact approximations by discarding some data from the original file, making it an irreversible compression method.
+Some compression methods such as {{glossary("WebP")}} are capable of both lossy and lossless compression depending on their usage and this compression mode is typically user-configurable.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/lossless_compression/index.md:12]: It seems slightly ambiguous what you mean by "this" in "this compression mode". Do you mean something like:

```suggestion
Some compression methods such as {{glossary("WebP")}} are capable of both lossy and lossless compression depending on their usage; the compression method used is typically user-configurable.
```

?

File: files/en-us/glossary/lossy_compression/index.md [hunk 3]
```diff
@@ -8,11 +8,12 @@ page-type: glossary-definition
 
 **Lossy compression**, or irreversible compression, is a data-compression method that uses inexact approximations and partial-data discarding to represent content. In simpler terms: lossy compression causes data from the initial file to be lost, possibly causing degradation in quality. The process of such compression is irreversible; once lossy compression of the content has been performed, the content cannot be restored to its original state. Therefore, content that has undergone lossy compression should generally not be further edited.
 
-Lossy compression is widely used in image formats.
+Lossy compression is widely used in image formats such as {{glossary("JPEG")}}, {{glossary("WebP")}}, and audio and video formats such as [MP3, MP4, H.264 and others](/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types).
+Some compression methods such as {{glossary("WebP")}} are capable of both lossy and lossless compression depending on their usage and this compression mode is typically user-configurable.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/lossy_compression/index.md:12]: Same comment as for the above instance.

File: files/en-us/glossary/lossy_compression/index.md [hunk 4]
```diff
@@ -8,11 +8,12 @@ page-type: glossary-definition
...
 
 ![Lossy compression image](2019-11-18.png)
 
-Although there is no obvious difference quality between the two images above, the size of the second image has been significantly reduced, using lossy compression.
+Although there is no obvious difference quality between the two images above, the size of the second image has been significantly reduced using lossy compression.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/lossy_compression/index.md:16]:
```suggestion
Although there is no obvious difference in quality between the two images above, the size of the second image has been significantly reduced using lossy compression.
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Nice changes @bsmth! A few comments for you, but nothing too major.

File: files/en-us/glossary/lossless_compression/index.md (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/glossary/lossless_compression/index.md:12]: Yup, sounds much better

**Review by chrisdavidmills — APPROVED** [review]
Your update suggestion sounds great. I think once you've applied that in both places, you can merge. Approving.

File: files/en-us/glossary/lossy_compression/index.md (hunk 3 above)
Comment (bsmth) [inline files/en-us/glossary/lossy_compression/index.md:12]:
```suggestion
Compression methods such as {{glossary("WebP")}} are capable of [both lossy and lossless compression](https://developers.google.com/speed/webp/docs/compression) depending on the compression level or the options you want to use during encoding.
```

Comment (bsmth) [reply files/en-us/glossary/lossy_compression/index.md]: Done 👍🏻

**Comment by bsmth** [comment]
> Your update suggestion sounds great. I think once you've applied that in both places, you can merge. Approving.

Nice one, thank you! 🙏🏻
